client.go

```diff
@@ -108,20 +108,27 @@ func (c *Client) invokePlugin(ctx context.Context, name string, r *types.Request
 	cmd.Stderr = os.Stderr
 
 	out, err := cmd.Output()
+	msg := "output:"
+	if len(out) > 0 {
+		msg = fmt.Sprintf("%s \"%s\"", msg, out)
+	} else {
+		msg = fmt.Sprintf("%s <empty>", msg)
+	}
 	if err != nil {
-		// ExitError is returned when there is a non-zero exit status
-		if _, ok := err.(*exec.ExitError); ok {
-			var pe types.PluginError
-			if err := json.Unmarshal(out, &pe); err != nil {
-				return nil, errors.Wrapf(err, "%s: %s", name, out)
-			}
-			return nil, &pe
+		if exitErr, ok := err.(*exec.ExitError); ok {
```

Comment:
`cmd.ProcessState.ExitCode()` can eliminate `if` branch ?

Reply:
Are you suggesting eliminating the `if` statement checking `exec.ExitError`? That allows us to return early if the plugin did not get a chance to run - say a general exec error due to the plugin binary missing or not being in ELF format. In those situations, I think, we want: `plugin: testplugin: exec: \"testplugin\": executable file not found in $PATH` (rather than trying to unmarshal the output and then reporting that there was a failure to unmarshal output).

Reply:
hm, i got the idea. I would rather use `LookPath` to make sure the binary exists before running before even executing the program.(I'd prefer to avoid nested if branches when possible for the sake of readability). But this one would work too.